tests/test_testRepo.py

```diff
@@ -0,0 +1,163 @@
+        self. _checkButlerDimension({"detector", "instrument"},
+                                    "detector=5",
+                                    [{"instrument": "notACam", "detector": 5},
+                                     {"instrument": "dummyCam", "detector": 5}])
+
+    def testAddDatasetType(self):
+        # 1 for StructuredDataNoComponents, 4 for StructuredData
+        self.assertEqual(len(self.butler.registry.getAllDatasetTypes()), 5)
+
+        # Testing the DatasetType objects is not practical, because all tests
+        # need a DimensionUniverse. So just check that we have the dataset
+        # types we expect.
+        self.butler.registry.getDatasetType("DataType1")
+        self.butler.registry.getDatasetType("DataType2")
+
+        with self.assertRaises(ValueError):
+            addDatasetType(self.butler, "DataType3", {"4thDimension"}, "NumpyArray")
+        with self.assertRaises(ValueError):
+            addDatasetType(self.butler, "DataType3", {"instrument"}, "UnstorableType")
+
+    def testRegisterMetricsExample(self):
+        id1 = {"instrument": "notACam"}
+        id2 = expandUniqueId(self.butler, {"visit": 101})
+        data = MetricsExample(summary={"answer": 42, "question": "unknown"})
+
+        self.butler.put(data, "DataType1", id1)
+        self.assertEqual(self.butler.get("DataType1", id1), data)
+
+        self.butler.put(data, "DataType2", id2)
+        self.assertEqual(self.butler.get("DataType2", id2), data)
+        self.assertEqual(self.butler.get("DataType2.summary", id2), data.summary)
+
+    def testUniqueButler(self):
+        dataId = {"instrument": "notACam"}
+        self.butler.put(MetricsExample({"answer": 42, "question": "unknown"}), "DataType1", dataId)
+        self.assertTrue(self.butler.datasetExists("DataType1", dataId))
+
+        newButler = makeTestCollection(self.creatorButler)
+        with self.assertRaises(LookupError):
+            newButler.datasetExists("DataType1", dataId)
+
+    def testExpandUniqueId(self):
+        self.assertEqual(dict(expandUniqueId(self.butler, {"instrument": "notACam"})),
+                         {"instrument": "notACam"})
+        self.assertIn(dict(expandUniqueId(self.butler, {"visit": 101})),
+                      [{"instrument": "notACam", "visit": 101},
+                       {"instrument": "dummyCam", "visit": 101}])
+        self.assertIn(dict(expandUniqueId(self.butler, {"detector": 5})),
+                      [{"instrument": "notACam", "detector": 5},
+                       {"instrument": "dummyCam", "detector": 5}])
+        self.assertIn(dict(expandUniqueId(self.butler, {"physical_filter": "k2020"})),
+                      [{"instrument": "notACam", "physical_filter": "k2020"},
+                       {"instrument": "notACam", "physical_filter": "k2020"}])
+        with self.assertRaises(ValueError):
+            expandUniqueId(self.butler, {"tract": 42})
+
+
+class MyMemoryTestCase(lsst.utils.tests.MemoryTestCase):
```

Comment:
I'm trying not to use lsst.utils.tests in daf_butler tests so please remove this and the init call below. (we have it in the Fits tests because those use afw and need to move at some point -- I also see one sneaked into the dbauth test). We are trying to be very careful about adding dependencies that we don't really need.